token_update_transaction_hip540_e2e_test.go

```diff
@@ -0,0 +1,718 @@
+//go:build all || e2e
+// +build all e2e
+
+package hedera
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+/*-
+ *
+ * Hedera Go SDK
+ *
+ * Copyright (C) 2020 - 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+// HIP-540 tests
+
+// KeyType represents different types of keys
+type KeyType int
+
+// Define enum values for different key types
+const (
+	WIPE_KEY KeyType = iota
+	KYC_KEY
+	SUPPLY_KEY
+	FREEZE_KEY
+	FEE_SCHEDULE_KEY
+	PAUSE_KEY
+	METADATA_KEY
+	ADMIN_KEY
+	ALL
+	LOWER_PRIVILEGE
+	NONE
+	ALL_DIFFERENT_ADMIN_KEY
+)
+
+const zeroKeyString = "0000000000000000000000000000000000000000000000000000000000000000"
```

Comment:
Maybe we should note to users that they can use this zeroKeyString from the HIP specification. The only place you can see this in action is this integration test suite. Maybe we can move the key to some other file in C++ for example I did something like this: 
```
/**
   * SDK needs to provide  a way to set an unusable key such as an Ed25519 all-zeros
   * key, since it is (presumably) impossible to find the 32-byte string whose SHA-512 hash begins with 32 bytes
   * of zeros. We recommend using all-zeros to clearly advertise any unsuable keys.
   */
  static inline const std::string ZERO_KEY_STR = "0000000000000000000000000000000000000000000000000000000000000000";
``` In the ED25519PrivateKey class.

Reply:
Added const ZERO_KEY_STRING